desk/src/components/List/ListFilters.vue

```diff
@@ -0,0 +1,190 @@
+						:options="allFilterableFields"
+						@update:model-value="(field) => addFilter(field)"
+					>
+						<template #target="{ togglePopover }">
+							<Button
+								class="!text-gray-600"
+								variant="ghost"
+								@click="togglePopover()"
+								label="Add Filter"
+							>
+								<template #prefix>
+									<FeatherIcon name="plus" class="h-3" />
+								</template>
+							</Button>
+						</template>
+					</Autocomplete>
+					<Button
+						v-if="filters?.length"
+						class="ml-auto !text-gray-600"
+						variant="ghost"
+						label="Clear All"
+						@click="clearFilters(close)"
+					>
+					</Button>
+				</div>
+			</div>
+		</template>
+	</NestedPopover>
+</template>
+
+<script setup>
+import ListFilterValue from "@/components/List/ListFilterValue.vue"
+import NestedPopover from "frappe-ui/src/components/ListFilter/NestedPopover.vue"
+
+import { Autocomplete } from "frappe-ui"
+import { getCurrentInstance } from "vue"
+import { useRoute, useRouter } from "vue-router"
+import {
+	linkTypes,
+	numberTypes,
+	stringTypes,
+	dateTypes,
+	filterOperators,
+} from "@/data/constants/filters"
+import { getFilterQuery } from "@/utils/list"
+
+const props = defineProps({
+	allFilterableFields: {
+		type: Array,
+		default: [],
+	},
+})
+
+const filters = defineModel()
+
+const getOperators = (index) => {
+	let f = filters.value[index]
+	let fieldtype = f.fieldtype
+
+	if (stringTypes.includes(fieldtype)) {
+		return filterOperators["string"]
+	} else if (numberTypes.includes(fieldtype)) {
+		return filterOperators["number"]
+	} else if (linkTypes.includes(fieldtype)) {
+		return filterOperators["link"]
+	} else if (dateTypes.includes(fieldtype)) {
+		return filterOperators["date"]
+	} else if (["Select", "Check", "Duration"].includes(fieldtype)) {
+		return filterOperators[fieldtype.toLowerCase()]
+	}
+
+	return []
+}
+
+const addFilter = (field) => {
+	filters.value.push({
+		fieldname: field.value,
+		fieldtype: field.type,
+		operator: "",
+		value: "",
+		options: field.options || [],
+	})
+}
+
+const removeFilter = (index) => {
+	filters.value.splice(index, 1)
+	updateFiltersInQuery()
+}
+
+const clearFilters = (close) => {
+	filters.value.splice(0, filters.value.length)
+	updateFiltersInQuery()
+	close
+}
+
+const updateField = (field, index) => {
+	filters.value[index] = {
+		fieldname: field.value,
+		fieldtype: field.type,
+		options: field.options || [],
+		operator: "",
+		value: "",
+	}
+}
+
+const updateOperator = (operator, index) => {
+	filters.value[index] = { ...filters.value[index], operator: operator, value: "" }
+}
+
+const updateValue = (value, operator, index) => {
+	let filter = filters.value[index]
+	let val = value.target ? value.target.value : value
+	if (operator === "between") {
+		val = val.split(",").map((v) => v.trim())
+	}
+	filter.value = val
+	updateFiltersInQuery()
+}
+
+const route = useRoute()
+const router = useRouter()
+const instance = getCurrentInstance()
```

Comment:
We should avoid using `getCurrentInstance`: https://v3.ru.vuejs.org/api/composition-api.html#getcurrentinstance

For calling grandparent's functions from grandchild or deeply nested components we can either:
- provide/inject functions
- Or maintain the functions and the state in the single store/reactive object and mutate the state from anywhere